merlin/analysis/filterbarcodes.py

```diff
@@ -395,7 +395,9 @@ def _run_analysis(self, fragmentIndex):
         bcDatabase = self.get_barcode_database()
         currentBarcodes = decodeTask.get_barcode_database()\
             .get_barcodes(fragmentIndex)
-        currentBarcodes = self.remove_z_duplicate_barcodes(currentBarcodes)
+
```

Comment:
This seems like it could change the "misidentification rate." Would it be possible to incorporate this into the calculation of the adaptive filter? I don't know the cleanest way to do this. Maybe the GenerateAdaptiveFilter task needs to know about the AdaptiveFilterBarcodes task so that it can read the value of remove_z_duplicate_barcodes. I don't want them to both have this parameter since it doesn't make sense to set this parameter indepedently.

Reply:
Yeah, I've thought about this. It does change the apparent misidentification rate for a given experiment, in my test cases it tended to increase the misidentification rate approximately proportional to the fraction removed as duplicates (i.e. if you removed 20% of your barcodes as apparent duplicates, the misidentification rate tended to go up about 20%), since it seems like blanks are depleted among the zplane duplicates. 

It would be most straightforward to address the duplicated barcodes at the decoding step, not the filtering step, and generate the adaptive threshold on the cleaned set of barcodes. It would add about a minute to the analysis of each z plane this way, which isn't bad. Would you prefer I implement it at decoding?

Reply:
That could work too. At first it seemed challenging since we decode and write one z plane at a time and wouldn't have enough information to remove duplicates, but I imagine we could read all the barcodes back in at the end and rewrite the barcodes with duplicates removed. Is this what only takes about a minute? If so this could be a good solution.

Reply:
Yeah, I figured I would just read in all the barcodes at the end and overwrite the original result. With a dataset that has 285 barcodes and 6 z planes, it takes about 3 minutes to remove duplicates when run on the full decoding result of a FOV. I think it should scale pretty linearly with number of decoded barcodes and the number of z planes you ask it to test against.